src/common/endoflife.py

```diff
@@ -74,3 +75,37 @@ def write_releases(product, releases, pathname="releases") -> None:
             # sort by date then version (desc)
             sorted(releases.items(), key=lambda x: (x[1], x[0]), reverse=True)
         ), indent=2))
+
+
+def parse_date(text, formats=frozenset([
+    "%B %d, %Y",  # January 1, 2020
+    "%b %d, %Y",  # Jan 1, 2020
+    "%B %d %Y",   # January 1 2020
+    "%b %d %Y",   # Jan 1 2020
+    "%d %B %Y",   # 1 January 2020
+    "%d %b %Y",   # 1 Jan 2020
+    "%d-%b-%Y",   # 1-Jan-2020
+    "%d-%B-%Y",   # 1-January-2020
+    "%B %Y",      # January 2020
+    "%b %Y",      # Jan 2020
+])): return parse_datetime(text, False, formats)
+
+
+def parse_datetime(text, to_utc=True, formats=frozenset([
```

Comment:
Move `to_utc` to the last parameter, so the signature is similar to `parse_date` with `formats` always second?

Could also make `to_utc` a named argument, so it must be called like:

```python
parse_datetime(my_text, my_format, to_utc=False
``` 

which is more explicit than:

```python
parse_datetime(my_text, my_formats, False
```